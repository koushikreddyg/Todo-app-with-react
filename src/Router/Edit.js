import React from 'react';
import { Col } from 'antd';
import 'antd/dist/antd.css';
import { Input } from 'antd';
import { Button } from 'antd';
import Actions from '../Library/Actions';
import editTask from '../store/Actions/editTask';
import removeTask from '../store/Actions/RemoveTask';
import { connect } from 'react-redux'
const Search = Input.Search;


export class Edit extends React.Component {
  
  constructor(props) {
    super(props);
    this.state = {
      editValue: Actions.validateId(this.props.Tasks,this.props.ID)===true?
      this.props.editTask1.task:'' ,
      id: this.props.ID,
      error:'',
    }
  }
  editValueChange = (e) => {
    e.preventDefault();
    const editValue = e.target.value;
    this.setState(() => ({
      editValue
    }))
  }
  submitEditValue = (e) => {
    e.preventDefault();
    const editValue = this.state.editValue;
    const id = this.state.id;
    const Value=editValue.trim();
    let error=Actions.errorCheck(Value,this.props.Tasks)
    if(Value.length===0){
      this.setState(()=>({
        error: 'please enter some value'
      }))
    }else if(error.length>0){
      this.setState(()=>({
        error: 'this task is in your task list'
      }))
    }else{
      this.setState(()=>({
        error:undefined
      }))
      this.props.editTask(id, { task: Value });
      this.setState(()=>({
        editValue:''
      }))
    this.props.history.push('/')
    }
    
  }
  Remove=(e)=>{
    e.preventDefault();
    this.props.removeTask(this.state.id);
    this.props.history.push('/');
  }
  componentDidMount(){
    if(this.state.editValue===''){
      this.props.history.push('/')
    }
  }

  render() {
    return (
      <div>
        (<Col offset={7}>
          <form onSubmit={this.submitEditValue}>
            <Search value={this.state.editValue}
              style={{ width: 400 }}
              placeholder='please enter the to do task' name='addInput'
              onChange={this.editValueChange} />
            <Button type='primary' onClick={this.submitEditValue}>Edit!</Button><br />
            {this.state.error&&<p>{this.state.error}</p>}
          </form>
          <Button type='primary' span={20} onClick={this.Remove}>Remove Me!</Button>
        </Col>
      </div>
    );
  }
}
const mapDispatchToProps=(dispatch)=>({
  removeTask:(id)=>dispatch(removeTask(id)),
  editTask:(id,object)=>dispatch(editTask(id,object)),
})
const mapStateToProps = (state, props) => ({

  editTask1: state.Actions.find((item) => (item.id === Number(props.match.params.id))),
  ID:Number(props.match.params.id),
  Tasks:state.Actions,
})
export default connect(mapStateToProps,mapDispatchToProps)(Edit);
