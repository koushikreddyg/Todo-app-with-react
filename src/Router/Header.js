import React from 'react';
import {NavLink} from 'react-router-dom';
 const Header=()=>(
  <div>
  <NavLink to="/" exact={true}>Home</NavLink>
  
  </div>
)
export default Header;