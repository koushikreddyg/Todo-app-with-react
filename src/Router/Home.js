import React from 'react';
import Form from '../components/Form';

class Home extends React.Component{
  constructor(props){
    super(props);
    this.state={
      options:[],
      error:false
    }
  }
Submit=(value)=>{
  const realvalue=value.trim();
  
  
}

render(){
  return(
    <div>
  
    <Form Submit={this.Submit}/>
    </div>
  )
}
}

export default Home;