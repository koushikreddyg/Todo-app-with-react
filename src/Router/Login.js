import React from 'react';
import {firebase,googleProvider,database} from '../firebase/firebase';
const Login=(props)=>(
    <div>
    <button onClick={(e)=>{
        firebase.auth().signInWithPopup(googleProvider).then(()=>{
            
        })
        
    }}>Login</button>
    </div>
)
export default Login