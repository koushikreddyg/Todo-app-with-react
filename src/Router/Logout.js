import React from 'react';
import {firebase,googleProvider,database} from '../firebase/firebase';
const Logout=(props)=>(
    <div>
    <button onClick={(e)=>{
        firebase.auth().signOut().then(()=>{
          
        })
       
    }}>Logout</button>
    </div>
)
export default Logout