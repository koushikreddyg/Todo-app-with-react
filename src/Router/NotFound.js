import React from 'react';
import {Link} from 'react-router-dom';
const NotFound=(props)=>(
  <div>
  This is not found page
  <Link to="/">Go Home</Link>
  </div>
)
export default NotFound;