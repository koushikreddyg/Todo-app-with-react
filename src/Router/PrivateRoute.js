import React from 'react';
import {Route,Redirect} from 'react-router-dom';
import {connect} from 'react-redux';
import Header from './Header';
const PrivateRoute=({
    isAuth,
    component:Component,
    ...rest
})=>(
    <div>
    <p>{isAuth}</p>
    <Route {...rest} component={(props)=>(
        isAuth?(<Component {...props}/>):(<Redirect to='/'/>)
        
    )}/>
    </div>
)
const mapStateToProps=(state)=>({
    isAuth:!!state.Auth.uid,
})
export default connect(mapStateToProps)(PrivateRoute);