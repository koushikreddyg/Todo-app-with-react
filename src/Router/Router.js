import React from 'react';
import Header from './Header';
import Edit from './Edit';
import NotFound from './NotFound';
import Home from './Home';
import Login from './Login'
import Logout from './Logout';
import {Router, Switch, Route} from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
export const history=createHistory()
 const AppRouter=()=>(
  <div>
  <Router history={history}>
  <div>
        <Switch>
          <Route path="/" component={Login} exact={true}/>
          <Route path="/dashboard" component={Home} />
          <Route path="/edit/:id" component={Edit}/>
          <Route  component={NotFound}/>
        </Switch>
        </div>
      </Router>
      
  
  </div>
)
export default AppRouter;