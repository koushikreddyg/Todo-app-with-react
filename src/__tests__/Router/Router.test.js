import React from 'react';
import Router from '../../Router/Router';
import {shallow} from 'enzyme';
test('should test router',()=>{
    const wrapper=shallow(<Router/>);
    expect(wrapper).toMatchSnapshot();
})