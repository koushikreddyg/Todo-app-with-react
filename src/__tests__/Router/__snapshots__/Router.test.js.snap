// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test router 1`] = `
<div>
  <BrowserRouter>
    <div>
      <Header />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/edit/:id"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </div>
  </BrowserRouter>
</div>
`;
